http2_general/test_h2_stream_priority.py

```diff
@@ -0,0 +1,691 @@
+            srv_group default {
+                server ${server_ip}:8000;
+            }
+            vhost good {
+                proxy_pass default;
+            }
+            tls_certificate ${tempesta_workdir}/tempesta.crt;
+            tls_certificate_key ${tempesta_workdir}/tempesta.key;
+            tls_match_any_server_name;
+
+            block_action attack reply;
+            block_action error reply;
+            http_chain {
+                host == "bad.com"   -> block;
+                                    -> good;
+            }
+        """
+    }
+
+    def test_max_concurent_stream_exceed_by_stream(self):
+        """
+        If creation of new stream leads to exceeding of max_concurrent_streams, we should reset
+        this stream. Check that all streams, which creation leads to exceedion of max_concurrent_streams
+        will be reset and all previos streams will be finished successfully.
+        """
+        client, server = self.setup_test_priority()
+
+        prev_stream_id = 0
+        self.assertTrue(client.h2_connection.remote_settings.max_concurrent_streams == 10)
+        for i in range(0, client.h2_connection.remote_settings.max_concurrent_streams):
+            stream_id = 2 * client.h2_connection.remote_settings.max_concurrent_streams + 2 * i + 1
+            client.send_bytes(
+                PriorityFrame(
+                    stream_id=stream_id,
+                    depends_on=prev_stream_id,
+                    stream_weight=1,
+                    exclusive=False,
+                ).serialize()
+            )
+            prev_stream_id = stream_id
+
+        valid_req_num = client.valid_req_num
+        time.sleep(1)
+        for i in range(0, client.h2_connection.remote_settings.max_concurrent_streams):
+            stream_id = client.stream_id
+            client.make_request(self.post_request)
+            self.assertTrue(client.wait_for_reset_stream(stream_id=stream_id))
+
+        client.reinit_hpack_encoder()
+        client.valid_req_num = valid_req_num
+        for i in range(0, client.h2_connection.remote_settings.max_concurrent_streams):
+            client.make_request(self.post_request)
+
+        self.wait_for_responses(client)
+        self.check_response_sequence(client, 10, [21, 23, 25, 27, 29, 31, 33, 35, 37, 39])
+
+    def test_max_concurent_stream_exceed_by_priority_frame(self):
+        """
+        If creation of new stream leads to exceeding of max_concurrent_streams, we should reset
+        this stream. But according to RFC we can't reset idle streams, so Tempesta FW just
+        close the connetion with PROTOCOL_ERROR.
+        """
+        client, server = self.setup_test_priority()
+
+        self.assertTrue(client.h2_connection.remote_settings.max_concurrent_streams == 10)
+        for i in range(0, client.h2_connection.remote_settings.max_concurrent_streams - 1):
+            client.make_request(self.post_request)
+        time.sleep(1)
```

Comment:
You should use. 
```suggestion
            client.make_request(self.post_request, end_stream=False)
```
Then you will receive 10 working streams.